python/src/scipp/table.py

```diff
@@ -160,270 +128,196 @@ def table_from_dataset(dataset,
-            if coord is not None:
-                html += _make_trailing_cells(dataset.labels, coord, i, row_end,
-                                             base_style, edge_style)
-                html += _make_trailing_cells(dataset.masks, coord, i, row_end,
-                                             base_style, edge_style)
-                html += _make_trailing_cells(dataset, coord, i, row_end,
-                                             base_style, edge_style)
+            # If there are bin edges, we need to add trailing cells
+            html += _make_trailing_cells(dict_of_variables, is_bin_centers, i,
+                                         row_end, base_style, edge_style)
             html += "</tr>"
+        # If we are not ending at the last row, add overflow
         if row_end != size:
-            html += _make_overflow_row(dataset, coord, base_style, hover_style)
+            html += _make_overflow_row(dict_of_variables, base_style,
+                                       hover_style)
 
     html += "</table>"
-    return html, size
+    return html
 
 
-def table(dataset):
+def table(scipp_obj):
     """
     Create a html table from the contents of a Dataset (0D and 1D Variables
-    only). The entries will be grouped by dimensions/coordinates.
+    only), DataArray, Variable or raw numpy array.
+    The entries will be grouped by dimensions/coordinates.
     """
 
     from IPython.display import display
-    tv = TableViewer(dataset)
+    tv = TableViewer(scipp_obj)
     display(tv.box)
 
 
 class TableViewer:
-    def __init__(self, dataset):
+    def __init__(self, scipp_obj):
 
+        # Delayed import
         self.widgets = __import__("ipywidgets")
 
-        self.tabledict = {
-            "default": sc.Dataset(),
-            "0D Variables": sc.Dataset(),
-            "1D Variables": {}
-        }
-        self.is_histogram = {}
+        groups = ["0D Variables", "1D Variables"]
+        self.tabledict = {}
+        self.is_bin_centers = {}
+        self.sizes = {}
+        for group in groups:
+            self.tabledict[group] = {}
+            self.is_bin_centers[group] = {}
+            self.sizes[group] = {}
         self.headers = 0
         self.trigger_update = True
-        is_empty = {}
-
-        tp = type(dataset)
-
-        if (tp is sc.Dataset) or (tp is sc.DatasetProxy):
 
+        if is_dataset_or_array(scipp_obj):
             self.headers = 2
+            if is_dataset(scipp_obj):
+                iterlist = scipp_obj.items()
+            else:
+                iterlist = [(scipp_obj.name, scipp_obj)]
+            # Note that capital 'D' is intentional on 'Data' to avoid the
+            # '.data' attribute of a DataArray being picked up
+            for field in ["coords", "labels", "masks", "Data"]:
+                if hasattr(scipp_obj, field):
+                    iter_items = getattr(scipp_obj, field).items()
+                else:
+                    iter_items = iterlist
```

Comment:
```python
for tag, field in zip(["coord", "labels", "mask", "data"], [x.coords, x.labels, x.masks, iterlist]):
```
less hacky and easier to read?

Reply:
yes this is better since it gets rid of the `getattr`